core/model-api/src/main/java/org/eclipse/rdf4j/model/base/AbstractLiteral.java

```diff
@@ -8,56 +8,88 @@
 
 package org.eclipse.rdf4j.model.base;
 
-import static java.lang.Boolean.FALSE;
-import static java.lang.Boolean.TRUE;
 import static java.util.Objects.requireNonNull;
 
 import java.math.BigDecimal;
 import java.math.BigInteger;
-import java.util.Locale;
-import java.util.Objects;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Optional;
 import java.util.function.Function;
-import java.util.regex.Pattern;
 
 import javax.xml.datatype.DatatypeConfigurationException;
+import javax.xml.datatype.DatatypeConstants;
 import javax.xml.datatype.DatatypeFactory;
 import javax.xml.datatype.XMLGregorianCalendar;
+import javax.xml.namespace.QName;
 
+import org.eclipse.rdf4j.model.IRI;
 import org.eclipse.rdf4j.model.Literal;
-import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.model.base.AbstractIRI.GenericIRI;
 
 /**
  * Base class for {@link Literal}, offering common functionality.
  *
  * @author Alessandro Bollini
  * @since 3.5.0
  */
+@SuppressWarnings("UseOfObsoleteDateTimeApi")
```

Comment:
Where does this come from? I've never seen this particular warning / annotation before.

Reply:
Technically speaking that annotation prevents IntelliJ automated QA from complaining about the use of `Date` and `Calendar`. Probably also useful as a human-readable reminder of the need to upgrade (e.g. https://github.com/eclipse/rdf4j/issues/2563).